clang/lib/CIR/Dialect/Transforms/TargetLowering/ItaniumCXXABI.cpp

```diff
@@ -67,6 +81,48 @@ bool ItaniumCXXABI::classifyReturnType(LowerFunctionInfo &FI) const {
   return false;
 }
 
+mlir::Type ItaniumCXXABI::getLLVMTypeForDataMember(
+    cir::DataMemberType type, mlir::TypeConverter &typeConverter) const {
+  // Itanium C++ ABI 2.3:
+  //   A pointer to data member is an offset from the base address of
+  //   the class object containing it, represented as a ptrdiff_t
+  const clang::TargetInfo &target = LM.getTarget();
+  unsigned width =
+      target.getTypeWidth(target.getPtrDiffType(clang::LangAS::Default));
```

Comment:
sounds like this method should be about returning width instead? Answering LLVM type questions at this point sounds like layer violation?